### PR TITLE
Filter "observatoryUri" output spam from test results (#1176).

### LINKED ISSUE #1176: Improve Flutter test console output.
The observatory link should open a browser on the given URL:

image

**EDIT**: full output also includes a line like:

    [{"event":"test.startedProcess","params":{"observatoryUri":"http://127.0.0.1:51770/"}}]

### PR DESCRIPTION
**Pull request by pq**
with flutter/flutter#11228, fixes #1176.

@devoncarew @stevemessick

### CODE REVIEW

**Comment by devoncarew** [comment]
@stevemessick is likely the best reviewer, given he wrote `DartTestEventsConverter`; just chipping in with some thoughts:

Do we know what it would take to make `DartTestEventsConverter` in the Dart plugin work for what we need here? In terms of how we'd need to parameterize its API? If so, can we track that work in an issue, introduce the API in the Dart plugin, and plan to remove this fork?

**Comment by stevemessick** [comment]
@devoncarew I'm on it. I just had to figure out how to download a patch so I can check the diff on that converter class.

File: third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/test/DartTestEventsConverterZ.java [hunk 1]
```diff
@@ -0,0 +1,825 @@
...
+    if (elem != null && elem.isJsonArray()) return process(elem.getAsJsonArray());
+
+    if (elem == null || !elem.isJsonObject()) return false;
+    return process(elem.getAsJsonObject());
+  }
+
+  /**
+   * Hook to process arrays.
+   */
+  abstract protected boolean process(JsonArray array);
```
Comment (stevemessick) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/test/DartTestEventsConverterZ.java]: Could just return false and be non-abstract.

Comment (pq) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/test/DartTestEventsConverterZ.java]: Done.

File: third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/test/DartTestEventsConverterZ.java [hunk 2]
```diff
@@ -0,0 +1,825 @@
...
+    if (group != null && group.getTestCount() > 0 && group.getDoneTestsCount() == group.getTestCount()) {
+      return processGroupDone(group) && checkGroupDone(group.getParent());
+    }
+    return true;
+  }
+
+  private boolean handleGroup(JsonObject obj) throws ParseException {
+    Group group = getGroup(obj.getAsJsonObject(DEF_GROUP));
+
+    System.out.println("handling group: " + group);
```
Comment (pq) [inline third_party/intellij-plugins-dart/src/com/jetbrains/lang/dart/ide/runner/test/DartTestEventsConverterZ.java]: Whoops!  Good catch.  Thanks!

**Comment by pq** [comment]
Thanks guys!

**Pull request merged**